### PR TITLE
Add table aws_vpc_vpn_connection #481

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT 300

SETUP: tests/aws_vpc_vpn_connection []

PRETEST: tests/aws_vpc_vpn_connection

TEST: tests/aws_vpc_vpn_connection
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_vpc.named_test_resource: Creating...
aws_customer_gateway.named_test_resource: Creating...
aws_vpc.named_test_resource: Still creating... [10s elapsed]
aws_customer_gateway.named_test_resource: Still creating... [10s elapsed]
aws_vpc.named_test_resource: Creation complete after 12s [id=vpc-0dd6571a0f172d3da]
aws_vpn_gateway.named_test_resource: Creating...
aws_customer_gateway.named_test_resource: Creation complete after 19s [id=cgw-08a7b029575ecfe30]
aws_vpn_gateway.named_test_resource: Still creating... [10s elapsed]
aws_vpn_gateway.named_test_resource: Still creating... [20s elapsed]
aws_vpn_gateway.named_test_resource: Creation complete after 27s [id=vgw-0cd2bef57eb8dcc07]
aws_vpn_connection.named_test_resource: Creating...
aws_vpn_connection.named_test_resource: Still creating... [10s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [20s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [30s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [40s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [50s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [1m0s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [1m10s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [1m20s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [1m30s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [1m40s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [1m50s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [2m0s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [2m10s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [2m20s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [2m30s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [2m40s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [2m50s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [3m0s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [3m10s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [3m20s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [3m30s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [3m40s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [3m50s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [4m0s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [4m10s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [4m20s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [4m30s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [4m40s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [4m50s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [5m0s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [5m10s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [5m20s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [5m30s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [5m40s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [5m50s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [6m0s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [6m10s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [6m20s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [6m30s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [6m40s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [6m50s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [7m0s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [7m10s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [7m20s elapsed]
aws_vpn_connection.named_test_resource: Still creating... [7m30s elapsed]
aws_vpn_connection.named_test_resource: Creation complete after 7m35s [id=vpn-0b0bc0a5e577f1a20]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

account_id = 123456789012
customer_gateway_id = cgw-08a7b029575ecfe30
resource_aka = arn:aws:ec2:us-east-1:123456789012:vpn-connection/vpn-0b0bc0a5e577f1a20
resource_id = vpn-0b0bc0a5e577f1a20
resource_name = turbottest8465
type = ipsec.1
vpn_gateway_id = vgw-0cd2bef57eb8dcc07

Running SQL query: test-get-query.sql
[
  {
    "customer_gateway_id": "cgw-08a7b029575ecfe30",
    "type": "ipsec.1",
    "vpn_connection_id": "vpn-0b0bc0a5e577f1a20",
    "vpn_gateway_id": "vgw-0cd2bef57eb8dcc07"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "customer_gateway_id": "cgw-08a7b029575ecfe30",
    "type": "ipsec.1",
    "vpn_connection_id": "vpn-0b0bc0a5e577f1a20",
    "vpn_gateway_id": "vgw-0cd2bef57eb8dcc07"
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "123456789012",
    "akas": [
      "arn:aws:ec2:us-east-1:123456789012:vpn-connection/vpn-0b0bc0a5e577f1a20"
    ],
    "tags": {
      "name": "turbottest8465"
    },
    "title": "vpn-0b0bc0a5e577f1a20"
  }
]
✔ PASSED

POSTTEST: tests/aws_vpc_vpn_connection

TEARDOWN: tests/aws_vpc_vpn_connection

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### VPN connection basic info

```sql
select
  vpn_connection_id,
  state,
  type,
  vpn_gateway_id,
  customer_gateway_id,
  region
from
  aws_vpc_vpn_connection;

+-----------------------+-----------+---------+-----------------------+-----------------------+-----------+
| vpn_connection_id     | state     | type    | vpn_gateway_id        | customer_gateway_id   | region    |
+-----------------------+-----------+---------+-----------------------+-----------------------+-----------+
| vpn-099f3d0ee361f108c | available | ipsec.1 | vgw-079b9e9f2d2411521 | cgw-0974df978d6e09c4b | us-east-1 |
| vpn-04db9c4b380497120 | deleted   | ipsec.1 | vgw-068c692e4f15c7fec | cgw-0cf522d67422b7b6a | us-east-1 |
| vpn-0e6b18be9be424b17 | deleted   | ipsec.1 | vgw-01f4243ea3a62d8ed | cgw-00d140cced7d1c918 | us-east-1 |
+-----------------------+-----------+---------+-----------------------+-----------------------+-----------+
```

### List VPN connection options info

```sql
select
  vpn_connection_id,
  options -> 'EnableAcceleration' as enable_acceleration,
  options ->> 'LocalIpv4NetworkCidr' as local_ipv4_network_cidr,
  options ->> 'LocalIpv6NetworkCidr' as local_ipv6_network_cidr,
  options ->> 'RemoteIpv4NetworkCidr' as remote_ipv4_network_cidr,
  options ->> 'RemoteIpv6NetworkCidr' as remote_ipv6_network_cidr,
  options -> 'StaticRoutesOnly' as static_routes_only,
  options ->> 'TunnelInsideIpVersion' as tunnel_inside_ip_iersion,
  options ->> '"TunnelOptions"' as "tunnel_options"
from
  aws_vpc_vpn_connection;

+-----------------------+---------------------+-------------------------+-------------------------+--------------------------+--------------------------+--------------------+--------------------------+----------------+
| vpn_connection_id     | enable_acceleration | local_ipv4_network_cidr | local_ipv6_network_cidr | remote_ipv4_network_cidr | remote_ipv6_network_cidr | static_routes_only | tunnel_inside_ip_iersion | tunnel_options |
+-----------------------+---------------------+-------------------------+-------------------------+--------------------------+--------------------------+--------------------+--------------------------+----------------+
| vpn-0e6b18be9be424b17 | false               | <null>                  | <null>                  | <null>                   | <null>                   | true               | <null>                   | <null>         |
| vpn-099f3d0ee361f108c | false               | 10.0.0.0/16             | <null>                  | 10.0.0.0/16              | <null>                   | false              | ipv4                     | <null>         |
| vpn-04db9c4b380497120 | false               | <null>                  | <null>                  | <null>                   | <null>                   | true               | <null>                   | <null>         |
+-----------------------+---------------------+-------------------------+-------------------------+--------------------------+--------------------------+--------------------+--------------------------+----------------+
```

### List VPN tunnel info of each VPN connection

```sql
select
  vpn_connection_id,
  t -> 'AcceptedRouteCount' as accepted_route_count,
  t ->> 'CertificateArn' as certificate_arn,
  t ->> 'LastStatusChange' as last_status_change,
  t ->> 'OutsideIpAddress' as outside_ip_address,
  t ->> 'Status' as status,
  t ->> 'StatusMessage' as status_message
from
  aws_vpc_vpn_connection,
  jsonb_array_elements(vgw_telemetry) as t;

+-----------------------+----------------------+-----------------+----------------------+--------------------+--------+----------------+
| vpn_connection_id     | accepted_route_count | certificate_arn | last_status_change   | outside_ip_address | status | status_message |
+-----------------------+----------------------+-----------------+----------------------+--------------------+--------+----------------+
| vpn-099f3d0ee361f108c | 0                    | <null>          | 2021-06-08T07:54:18Z | 34.205.109.112     | DOWN   | IPSEC IS DOWN  |
| vpn-099f3d0ee361f108c | 0                    | <null>          | 2021-06-08T07:59:59Z | 50.17.255.86       | DOWN   | IPSEC IS DOWN  |
+-----------------------+----------------------+-----------------+----------------------+--------------------+--------+----------------+
```
```
</details>

### CODE REVIEW

File: aws/table_aws_vpc_vpn_connection.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+				Description: "The configuration information for the VPN connection's customer gateway.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "transit_gateway_id",
+				Description: "The ID of the transit gateway associated with the VPN connection.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			// JSON columns
```
Comment (Subhajit97) [inline aws/table_aws_vpc_vpn_connection.go:77]:
```suggestion
```

File: aws/table_aws_vpc_vpn_connection.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromP(getVpcVpnConnectionTurbotData, "Tags"),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(getVpcVpnConnectionTurbotData, "Title"),
+			},
```
Comment (Subhajit97) [inline aws/table_aws_vpc_vpn_connection.go:112]: Add `title` column before `tags`

File: aws/table_aws_vpc_vpn_connection.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags that are attached to VPN gateway.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			//standard columns
```
Comment (Subhajit97) [inline aws/table_aws_vpc_vpn_connection.go:100]:
```suggestion
			// Steampipe standard columns
```

File: docs/tables/aws_vpc_vpn_connection.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  t -> 'AcceptedRouteCount' as accepted_route_count,
+  t ->> 'CertificateArn' as certificate_arn,
+  t ->> 'LastStatusChange' as last_status_change,
+  t ->> 'OutsideIpAddress' as outside_ip_address,
+  t ->> 'Status' as status,
+  t ->> 'StatusMessage' as status_message
+from
+  aws_vpc_vpn_connection,
+  jsonb_array_elements(vgw_telemetry) as t;
+```
```
Comment (Subhajit97) [inline docs/tables/aws_vpc_vpn_connection.md:54]: @rajeshbal65 Can we add a query to check the status of both VPN tunnels (both should be in `UP` state)?

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@rajeshbal65 There are few changes required. Please see comments, thanks!

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**